ddtrace/propagation/http.py

```diff
@@ -747,15 +756,23 @@ def _extract(headers):
             if tp is None:
                 log.debug("no traceparent header")
                 return None
-            trace_id, span_id, sampling_priority = _TraceContext._get_traceparent_values(tp)
+            trace_id, span_id, trace_flag = _TraceContext._get_traceparent_values(tp)
         except (ValueError, AssertionError):
             log.exception("received invalid w3c traceparent: %s ", tp)
             return None
-        origin = None
+
         meta = {W3C_TRACEPARENT_KEY: tp}  # type: _MetaDictType
 
         ts = _extract_header_value(_POSSIBLE_HTTP_HEADER_TRACESTATE, headers)
+        return _TraceContext._get_context(trace_id, span_id, trace_flag, ts, meta)
 
+    @staticmethod
+    def _get_context(trace_id, span_id, trace_flag, ts, meta=None):
```

Comment:
Refactor so we can get a context object with supplying distributed tracing headers. This will allow us to reuse this propagation logic in the otel api

Reply:
No functionality changed here. I just renamed some variables and make this code slightly easier to read

Reply:
@emmettbutler  This refactor introduced the 2% performance regression